utils/gyb_syntax_support/AttributeKinds.py

```diff
@@ -861,7 +867,7 @@ def __init__(self, name, swift_name=None):
                                   code=60),
 
     ContextualSimpleDeclAttribute('isolated', 'Isolated',
-                                  DeclModifier, OnParam,
+                                  DeclModifier, OnParam, OnDestructor,
```

Comment:
Given that the parser needed to be updated to accept `isolated` on destructors, we need to be careful about interface printing. ASTPrinter.cpp needs to be updated with a new "feature" for isolated deinits and it needs to print the declaration of any `deinit` with `isolated` with a feature guard so that older development versions of the compiler will continue to be able to parse `swiftinterfaces` for modules that have adopted this feature. There should also be a test under the `tests/ModuleInterface/`subdirectory that exercises printing of isolated deinits and verifies that they are printed with the feature guards.

Reply:
`isolated` is an attribute, and there is already code in place in ASTPrinter which prints attributes, so it should work, but worth to add couple more tests cases. But I could not find which test tests ASTPrinter. Any examples to look at?

Reply:
`tests/ModuleInterface/` contains examples of tests that exercise printing.

To be clearer about what I'm saying the problem will be: already existing compilers that do not have the changes from this PR will presumably reject this declaration when parsing a `.swiftinterface`:
```
class SomeClass {
  isolated deinit // error: 'isolated' may only be used on 'parameter' declarations
}
```
Therefore, to maintain compatibility with older compilers, we need to print the deinit declaration that has adopted this new capability something like this:
```
class SomeClass {
  #if $IsolatedDeinit
  isolated deinit
  #else
  deinit
  #endif
}
```
This can be achieved with a suppressible language feature (see https://github.com/apple/swift/blob/main/include/swift/Basic/Features.def)

Reply:
Perhaps, yes. But that gets tricky because other references to the class elsewhere in the interface would also need to be guarded, making it viral. The ability to parse the interface successfully is the most important aspect of this. Preventing issues with the code generated by the older compiler is definitely important but secondary as usually a client building with the older compiler are not using any of the newer declarations from the newer interface. It's a temporary configuration that we're just trying to make sure is possible to use for clients with existing, shipping code.

Reply:
> But that gets tricky because other references to the class elsewhere in the interface would also need to be guarded, making it viral.

Not sure if I understood this bit.

I meant something like this:
```swift
#if $IsolatedDeinit
open class SomeClass {
  isolated deinit
}
#else
public class SomeClass {
}
#endif

func foo(_ x: SomeClass)
```

For example, why would reference in `foo()` need to be guarded?

Reply:
@tshortli, implemented, please take a look

Reply:
> Not sure if I understood this bit.

Sorry, what I said didn't make sense because I read too quickly and thought you were going to hide the class definition of the class entirely when the feature is unavailable, in which case you have to consider what happens to references to the class in other declarations. The approach of preventing subclassing when the feature is unavailable is a clever one that should work nicely here.

I'll take a look at the updates to the PR in more depth tomorrow.